### PR TITLE
Various small time-related test improvements

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Comment by wthrowe** [comment]
What is an ineffective cast?

**Comment by kidder** [comment]
Yeah, I find that wording to be confusing.  Trying to interpret [the description of the failing clang-tidy check](https://clang.llvm.org/extra/clang-tidy/checks/bugprone-misplaced-widening-cast.html),  it seems by ineffective, they mean that the result of the arithmetic operation would give you the expected answer in the narrower type (and therefore it was unnecessary to do a widening cast).  But the cast obviously has an effect as it changes the type so ineffective seems a poor choice of wording...   To me it seems the point of the check is that it is safer to perform the widening cast on one of the operands instead of on the entire arithmetic operation, and they should just say that...

**Comment by wthrowe** [comment]
Filed https://bugs.llvm.org/show_bug.cgi?id=48861 upstream.

File: tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp [hunk 1]
```diff
@@ -482,22 +486,26 @@ void check_dense_output(const TimeStepper& stepper,
...
+      TimeSteppers::History<double, double> history;
+      const auto rhs = [](const double v, const double /*t*/) noexcept {
+        return v;
+      };
+      initialize_history(
+          time, &history, [](const double t) noexcept { return exp(t); }, rhs,
+          time_step, stepper.number_of_past_steps());
+      take_step(&time, &y, &history, stepper, rhs, time_step);
+
+      CHECK(get_dense(time_step, std::numeric_limits<double>::epsilon() *
```
Comment (moxcodes) [inline tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp:505]: These epsilons are to ensure that the dense output doesn't land directly on a step value, right? Could you add a code comment explaining that?

**Review by moxcodes — DISMISSED** [review]
Looks good, just one place where I thought a comment could be helpful in one of the tests. Feel free to squash that directly.

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
I have no comments in addition to @moxcodes's comment

File: tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp (hunk 1 above)
Comment (wthrowe) [reply tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp:505]: Yes, but it's complicated.  The "correct" test tests 0, eps, 1-eps, and 1, but 0 and 1 (actually I think just 0) are broken in at least one stepper.  I will fix that, but I want this PR to be only changes to the tests, so I can't fix it here.  Do you still want a comment?

Comment (moxcodes) [reply tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp:505]: I think a comment would be good -- I just feel it's helpful  for reading and maintaining the test to understand that the `eps` case actually acts differently in the implementation from the `0` case. I think that's not necessarily obvious without understanding the dense-output mathematics.

Comment (wthrowe) [reply tests/Unit/Helpers/Time/TimeSteppers/TimeStepperTestUtils.cpp:505]: Good point.  I will add a comment to that effect.

**Review by moxcodes — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**